diesel/src/types/impls/date_and_time/chrono.rs

```diff
@@ -45,6 +47,38 @@ impl ToSql<Timestamp> for NaiveDateTime {
     }
 }
 
+fn midnight() -> NaiveTime {
+    NaiveTime::from_hms(0, 0, 0)
+}
+
+impl ToSql<Time> for NaiveTime {
+    fn to_sql<W: Write>(&self, out: &mut W) -> Result<IsNull, Box<Error>> {
+        let duration = *self - midnight();
+        match duration.num_microseconds() {
+            Some(offset) => ToSql::<Time>::to_sql(&PgTime(offset), out),
+            None => {
+                let error_message = format!("{:?} as microseconds is too large to fit in an i64",
+                                            self);
+                return Err(Box::<Error + Send + Sync>::from(error_message));
+            }
+        }
+    }
+}
+
+impl FromSql<Time> for NaiveTime {
+    fn from_sql(bytes: Option<&[u8]>) -> Result<Self, Box<Error>> {
+        let PgTime(offset) = try!(FromSql::<Time>::from_sql(bytes));
+        if offset >= 0 {
+            let duration = Duration::microseconds(offset);
+            Ok(midnight() + duration)
+        } else {
+            let error_message = format!("Negative time: {:?} (time should always be positive)",
```

Comment:
I think this is just `unreachable!("Received negative time from PG (time should always be positive)")`